web/ui/src/lib/clustersMap/campus/lyon.ts

```diff
@@ -0,0 +1,97 @@
+import { Campus } from '../campus';
+import { Cluster } from '../cluster';
+import { ICampus } from '../types';
+
+//
+export class Lyon extends Campus implements ICampus {
+  emoji = (): string => '🇫🇷';
+
+  name = (): string => 'Lyon';
+
+  extractorRegexp = (): RegExp =>
+    /(?<clusterWithLetter>(?<cluster>\d+))(?<rowWithLetter>(?<row>\w))(?<workspaceWithLetter>(?<workspace>\d+))/i;
```

Comment:
This is the logic for the regexp, you just two things : 

1. Cluster with Letter:
Now it explicitly requires the letter "Z" before capturing the digits in the clusterWithLetter group. This means the string must start with "Z" followed by numbers (e.g., "Z123").

2. Row with Letter:
Now the row part only allows digits, represented by `\d+`. This change ensures that the row can only be numeric (e.g., "456"), whereas the previous version would have allowed both letters and numbers (e.g., "A12").

```suggestion
    /(?<clusterWithLetter>z(?<cluster>\d+))(?<rowWithLetter>r(?<row>\d+))(?<workspaceWithLetter>p(?<workspace>\d+))/i;
```